src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -927,7 +951,8 @@ public void Dispose()
         // Handle Exited event and display process information.
         private void myProcess_Exited(object sender, System.EventArgs e)
         {
-            if (System.Threading.Interlocked.Decrement(ref _numberOfProcessesToWaitFor) == 0)
+            int decremented = System.Threading.Interlocked.Decrement(ref _numberOfProcessesToWaitFor);
+            if ((decremented == 0) || (Any && decremented < _initialNumberOfProcesses))
```

Comment:
Why do we need _initialNumberOfProcesses? If I understand correctly first terminated process must call `_waitHandle?.Set();`.